src/forms/TextInput.jsx

```diff
@@ -5,18 +5,14 @@ import cx from 'classnames';
 /**
  * @module TextInput
  */
-class TextInput extends React.Component {
+class TextInput extends React.PureComponent {
 
 	constructor(props) {
 		super(props);
-		this.state = {
-			value: props.value || '',
-		};
 		this.onChange = this.onChange.bind(this);
 	}
 
 	onChange(e) {
-		this.setState({ value: e.target.value });
 		if (this.props.onChange) {
 			this.props.onChange(e);
```

Comment:
I agree with @mikespencer we should be trying to go for stateless functional components as much as possible. Especially at the MWC level. I think my only concern is how this change would interact if we wanted this component to work with _redux forms_. @mikespencer any concerns there?